frontend/src/AutocompleteWrapper.ts

```diff
@@ -1,206 +1,144 @@
-
-  constructor({ appId, apiKey, siteId, branch }: Options) {
-    this.client = this.createClient(appId, apiKey);
-    const indexName = this.computeIndexName(siteId, branch);
-    this.index = this.client.initIndex(indexName);
+  private options;
+  private indexName;
+  private client;
+  private autocomplete: AutocompleteApi<AlgoliaRecord> | undefined;
+
+  constructor(options: Options) {
+    this.options = options;
+    this.client = this.createClient();
+    this.indexName = this.computeIndexName();
   }
 
-  render({
-    analytics,
-    autocomplete: { hitsPerPage, inputSelector },
-    color,
-    debug,
-    silenceWarnings,
-    poweredBy,
-  }: Options) {
-    addCss(templates.autocomplete.css(color));
-
-    const $inputs = this.getInputs(inputSelector);
-
-    if ($inputs.length === 0 && !silenceWarnings) {
-      const inputSelectorText = JSON.stringify(inputSelector);
-      console.warn(
-        [
-          `[Algolia] No input matched our default selector ${inputSelectorText}`,
-          'The integration needs a search input to be active on your page. You can either:',
-          '- add an input that matches the selector',
-          '- or modify `autocomplete.inputSelector` to match your search input.',
-        ].join('\n')
-      );
+  render() {
+    const $input = document.querySelector(this.options.selector) as HTMLElement;
+    if (!$input) {
+      console.error('[algoliasearch netlify] no inputs found');
+      return;
     }
 
-    const autocompletes = $inputs.map(($input) => {
-      const inputWidth = $input.getBoundingClientRect().width;
-
-      const sizeModifier = this.computeSizeModifier(inputWidth);
-      const nbSnippetWords = this.computeNbSnippetWords(inputWidth);
-
-      const autocompleteParams = {
-        hint: false,
-        debug,
-        templates: this.getDropdownTemplates(poweredBy),
-        appendTo: 'body',
-      };
-
-      const searchParams = {
-        analytics,
-        hitsPerPage,
-        highlightPreTag: '<span class="aa-hit--highlight">',
-        highlightPostTag: '</span>',
-        attributesToSnippet: [
-          `description:${nbSnippetWords}`,
-          `content:${nbSnippetWords}`,
-        ],
-        snippetEllipsisText: '...',
-      };
-
-      const sources = [
-        {
-          name: 'hits',
-          source: this.createSource(searchParams),
-          templates: {
-            suggestion: this.createRenderSuggestion(sizeModifier),
-          },
-        },
-      ];
-
-      const aa: AutocompleteJs = autocomplete(
-        $input,
-        autocompleteParams,
-        sources
-      );
-
-      const handleSelected = this.createHandleSelected();
-      aa.on('autocomplete:selected', handleSelected);
-
-      return aa;
+    const instance = autocomplete<AlgoliaRecord>({
+      container: $input,
+      autoFocus: false,
+      placeholder: 'Search...',
+      debug: this.options.debug,
+      openOnFocus: true,
```

Comment:
This one too please, I personally don't like to see random results before I type anything

Reply:
As a recommendation, `openOnFocus` works well when you want to display recent searches, hot searches or a site map. I wouldn't use it by default in this integration.